staking-pool/src/lib.rs

```diff
@@ -183,6 +183,10 @@ impl StakingContract {
             "The owner account ID is invalid"
         );
         let account_balance = env::account_balance();
+        assert!(
+            account_balance > STAKE_SHARE_PRICE_GUARANTEE_FUND,
```

Comment:
Why this is strictly higher?

Reply:
because if `account_balance == STAKE_SHARE_PRICE_GUARANTEE_FUND` then in the next line total_staked_balance becomes zero. `total_staked_balance` is used in several share calculations as denominator, so `total_staked_balance==0` leads to div by zero panics. (i.e. `fn num_shares_from_staked_amount_rounded_down/up`)